src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -783,6 +783,15 @@ abstract class BTypes {
       } while (fcs == null)
       fcs
     }
+
```

Comment:
hashcode and equls are needed by one of the unit tests, and as this is no longer a case class neede to be explicit

Reply:
I'm in favor of removing the overrides, or is it too hard to adjust the tests?

Reply:
probably not hard - will find out which tests failed

Reply:
failed test was 
```
[info] Test scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.compareClassBTypes started
[error] Test scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.compareClassBTypes failed: java.lang.AssertionError: assertion failed: Ljava/lang/Object; != Ljava/lang/Object;, took 0.014 sec
[error]     at scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.sameBType(BTypesFromClassfileTest.scala:39)
[error]     at scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.$anonfun$check$1(BTypesFromClassfileTest.scala:106)
[error]     at scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.duringBackend(BTypesFromClassfileTest.scala:26)
[error]     at scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.check(BTypesFromClassfileTest.scala:94)
[error]     at scala.tools.nsc.backend.jvm.opt.BTypesFromClassfileTest.compareClassBTypes(BTypesFromClassfileTest.scala:113)
[error]     ...

```

Reply:
and should be fixed by https://github.com/scala/scala/pull/6201/commits/261dde55d1b3b4ac87bfa948cd6960feba698bec